integration/tests/cook/test_cli.py

```diff
@@ -858,6 +858,41 @@ def test_tail_wait(self):
         self.assertEqual(1, cp.returncode, cli.decode(cp.stderr))
         self.assertEqual(f"Cannot open '{path}' for reading (file was not found).\n", cli.decode(cp.stderr))
 
+    def test_tail_with_plugin(self):
+        # User defined plugin to print dummy content from a file
+        with tempfile.NamedTemporaryFile(suffix='.py', delete=True) as temp:
+            plugin_code = """
+def dummy_tail_text(instance, sandbox_dir, path, offset=None, length=None):
+    if offset is None:
+        return {'data': '', 'offset': 12}
+    else:
+        return {'data': 'Hello\\nworld!', 'offset': 0}
+"""
+            temp.write(plugin_code.encode())
+            temp.flush()
+            config = {
+                'plugins': {
+                    "tail-plugin": {
+                        "module-name": "does_not_matter",
+                        "path": temp.name,
+                        "function-name": "dummy_tail_text"
+                    }
+                }
+            }
+
+            with cli.temp_config_file(config) as path:
+                flags = f'--config {path} --verbose'
+                cp, uuids = cli.submit('touch file.txt', self.cook_url)
+                self.assertEqual(0, cp.returncode, cp.stderr)
+                cp = cli.wait(uuids, self.cook_url)
+                self.assertEqual(0, cp.returncode, cp.stderr)
+                cp = cli.tail(uuids[0], 'file.txt', self.cook_url, flags=flags)
```

Comment:
Do we even need to submit a job to test this?

Reply:
We need to submit a job to get a uuid to pass into `tail`, and `touch` creates a file to pass into `tail` as well.

Reply:
Do we have to `wait` for the job?

Reply:
I'm not sure if it's completely necessary, but the other `tail` integration tests `wait` until the job is completed, so I thought I would do the same, even if it is just creating a file.

Reply:
If we can skip the `wait` and the test still works, I think it's worth it to make the test faster and potentially more reliable.